### PR TITLE
2 dashboard view profile

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### Changes Implemented:
  - #25 

### Quality Control Checklist:

  - [X] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
  - [ ] 100% SimpleCov test coverage (if below, please clarify below)
    - 158/161 (98.14%)
  - [X] Last commit has passing Travis build

### Blockers (if applicable):
  - It is starting to get tricky to test with the endpoints.  In particular, with other features that visited the dashboard, I needed to stub out the API calls in the `dashboard` action of the `users` controller.

### Next Steps & Additional Notes:
 - The profile page needs some test love.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -20,7 +20,21 @@ def dashboard
...
 
-  def profile; end
+  def profile
+    current_user_friends = BackEndFacade.get_user_friends(current_user.id)
+    @user = BackEndFacade.get_user(params[:user_id])
+    @user_friends = BackEndFacade.get_user_friends(@user.id)
+    @user_gyms = BackEndFacade.get_user_gyms(@user.id)
+    @user_events = BackEndFacade.get_user_events(@user.id)
+
+    @user_profile = if current_user.id.to_s == params[:user_id]
```
Comment (gunnarrunner) [inline app/controllers/users_controller.rb:30]: Thank you so much for adding this button for just the current users page.

File: app/views/users/profile.html.erb [hunk 2]
```diff
@@ -1,3 +1,79 @@
 <% if current_user %>
-  <%= render partial: 'profiles', locals: { view_type: 'profile (self)' } %>
+  <table class="table table-sm table-borderless table-responsive align-middle">
+    <tr>
+      <td align="center">
+        <div id='profile'>
+          <img id="profile_pic" src="<%= @user.google_image_url %>"><br>
+          <% if @user_profile == :self %>
```
Comment (gunnarrunner) [inline app/views/users/profile.html.erb:7]: Awesome good work!

File: spec/features/users/profile_spec.rb [hunk 3]
```diff
@@ -1,8 +1,13 @@
 require 'rails_helper'
 RSpec.describe 'user profile page' do
   it 'can show the attributes of the current user', :vcr do
+    allow(BackEndFacade).to receive(:get_user_friends).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_gyms).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_events).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user).with(@user.id.to_s).and_return(@user)
```
Comment (Scott-Borecki) [inline spec/features/users/profile_spec.rb:7]: These stubs needed to be added to get the profile spec to pass.

File: spec/features/users/register_spec.rb [hunk 4]
```diff
@@ -22,6 +22,10 @@
     end
 
     it 'can register a new user if all required attributes are provided', :vcr do
+      allow(BackEndFacade).to receive(:get_user_friends).with(@user.id).and_return([])
+      allow(BackEndFacade).to receive(:get_user_gyms).with(@user.id).and_return([])
+      allow(BackEndFacade).to receive(:get_user_events).with(@user.id).and_return([])
```
Comment (Scott-Borecki) [inline spec/features/users/register_spec.rb:27]: These stubs needed to be added to get the register spec to pass.

File: spec/features/welcome/index_spec.rb [hunk 5]
```diff
@@ -12,6 +12,10 @@
   end
 
   it 'can log in and out of the application with a valid Google token', :vcr do
+    allow(BackEndFacade).to receive(:get_user_friends).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_gyms).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_events).with(@user.id).and_return([])
```
Comment (Scott-Borecki) [inline spec/features/welcome/index_spec.rb:17]: These stubs needed to be added to get the welcome index spec to pass.

File: spec/features/gym_search/index_spec.rb [hunk 6]
```diff
@@ -1,7 +1,10 @@
 require 'rails_helper'
 
-RSpec.describe 'gyms near me page' do
+RSpec.describe 'gyms near me page', :vcr do
   before do
+    allow(BackEndFacade).to receive(:get_user_friends).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_gyms).with(@user.id).and_return([])
+    allow(BackEndFacade).to receive(:get_user_events).with(@user.id).and_return([])
```
Comment (Scott-Borecki) [inline spec/features/gym_search/index_spec.rb:7]: These stubs needed to be added to get the gym search index spec to pass.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (tvaroglu) [inline app/controllers/users_controller.rb:30]: Conditional SWAG

**Review by gunnarrunner — APPROVED** [review]
Great work. Thank you for adding the edit profile button for only the current users profile saves me the headache!

File: spec/features/gym_search/index_spec.rb (hunk 6 above)
Comment (tvaroglu) [reply spec/features/gym_search/index_spec.rb:7]: Adding on my end as we speak

File: spec/features/users/profile_spec.rb (hunk 3 above)
Comment (tvaroglu) [reply spec/features/users/profile_spec.rb:7]: Added on my branch, same problem

File: spec/features/users/register_spec.rb (hunk 4 above)
Comment (tvaroglu) [reply spec/features/users/register_spec.rb:27]: Added on my branch, same problem

File: spec/features/welcome/index_spec.rb (hunk 5 above)
Comment (tvaroglu) [reply spec/features/welcome/index_spec.rb:17]: Added on my branch, same problem

**Review by tvaroglu — APPROVED** [review]
Like a boss!  Thank you

Approved

**Pull request merged**